Quadriga/src/main/java/edu/asu/spring/quadriga/web/sites/PublicNetworkController.java

```diff
@@ -1,56 +1,97 @@
 
+    @Autowired
+    private IQStoreAsyncTaskManager qStoreAsyncTaskManager;
+
     /**
-     * This method loads all the public networks and navigates to the public
-     * networks page to display the graph
+     * This method navigates to the public networks page to display the graph
      * 
-     * @param model
      * @return
-     * @throws QuadrigaStorageException
      */
     @RequestMapping(value = "sites/network", method = RequestMethod.GET)
-    public String showPublicNetworks(Model model) throws QuadrigaStorageException {
-        List<INetwork> networkList = networkManager.getApprovedNetworkList();
-        ITransformedNetwork transformedNetwork = transformationManager.getTransformedApprovedNetworks(networkList);
+    public String showPublicNetworks() {
+
+        return "sites/network";
+    }
+
+    /**
+     * This method returns a JSON String that represents all the public networks
+     * 
+     * @throws QStoreStorageException
+     * @throws JAXBException
+     * @throws QuadrigaStorageException
+     * 
+     */
+    @RequestMapping(value = "public/rest/networks", method = RequestMethod.GET)
+    public ResponseEntity<String> getPublicNetwork()
+            throws QStoreStorageException, QuadrigaStorageException, JAXBException {
+        String xml = qStoreAsyncTaskManager.getNetworkWithPopularTerms();
+        ITransformedNetwork transformedNetwork = transformationManager.getAllTransformedNetworks(xml);
 
         String json = null;
-        if (transformedNetwork != null) {
+        if (transformedNetwork != null && transformedNetwork.getNodes().size() != 0) {
             json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
         }
 
-        if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
-            model.addAttribute("isNetworkEmpty", true);
+        return new ResponseEntity<>(json, HttpStatus.OK);
+    }
+
+    /**
+     * This method starts asynchronous query execution
+     * 
+     * @return
+     */
+    @RequestMapping(value = "public/rest/loadnetworks", method = RequestMethod.GET)
+    public ResponseEntity<String> loadNetworks() {
+        try {
+            qStoreAsyncTaskManager.startLoadingPublicNetworks();
+        } catch (QStoreStorageException e) {
+            return new ResponseEntity<>(FAILED.name(), HttpStatus.INTERNAL_SERVER_ERROR);
         }
+        return new ResponseEntity<>(RUNNING.name(), HttpStatus.OK);
+    }
 
-        model.addAttribute("jsonstring", json);
+    /**
+     * This method tells if the asynchronous query has loaded the public
+     * networks or not
+     * 
+     * @return
+     */
+    @RequestMapping(value = "public/rest/networkloadstatus", method = RequestMethod.GET)
+    public ResponseEntity<String> getPublicNetworkLoadStatus() {
+        if (qStoreAsyncTaskManager.getPublicNetworkStatus()) {
+            return new ResponseEntity<>(COMPLETED.name(), HttpStatus.OK);
```

Comment:
what if there was an error in QStore? (but no exception, just an error message coming back from qstore)